src/bottom_topography/spline_utils_2D.jl

```diff
@@ -0,0 +1,123 @@
+# Helpfunctions for 2D spline interpolation
+
+# Sort data to follow logic of the remaining code.
+# x and y are sorted ascending. The matrix z contains
+# the corresponding values.
+function sort_data(x::Vector{Float64}, y::Vector{Float64}, z::Matrix{Float64})
+
+  zx              = transpose(z)
+  original_data_x = hcat(x, zx)
+  sorted_data_x   = original_data_x[sortperm(original_data_x[:,1]), :]
+
+  x_sorted  = sorted_data_x[:,1]
+  z_interim = sorted_data_x[:, 2:end]
+
+  zy              = transpose(z_interim)
+  original_data_y = hcat(y, zy)
+  sorted_data_y   = original_data_y[sortperm(original_data_y[:,1]), :]
+
+  y_sorted = sorted_data_y[:,1]
+  z_sorted   = sorted_data_y[:,2:end]
+  
+  return x_sorted, y_sorted, Matrix(z_sorted)
+end
+
+###################################
+### Thin plate spline smoothing ###
+###################################
+
+# Base function for thin plate spline
+function tps_base_func(r)
+
+  if r == 0
+    return 0
+  else
+    return r*r*log(r)
+  end
+    
+end
+
+# restructure input data to be able to use the thin plate spline functionality
+# data is restructured in the following form:
+# x = (x_1, x_2, ..., x_n)
+# y = (y_1, y_2, ..., y_m)
+# z = (z_11, ..., z_1n;
+#      ...., ..., ....;
+#      z_nm, ..., z_nm)
+# to control_points = (x_1, y_1, z_11;
+#                      x_2, y_1, z_21;
+#                      ..., ..., ....;
+#                      x_n, y_m, z_nm)
+function restructure_data(x, y, z)
+
+  x_mat = repeat(x, 1, length(y))
+  y_mat = repeat(y, 1, length(x))
+  
+  p = length(z)
+
+  x_vec = vec(reshape(x_mat , (p, 1)))
+  y_vec = vec(reshape(y_mat', (p, 1)))
+  z_vec = vec(reshape(z'    , (p, 1)))
+
+  return [x_vec y_vec z_vec]
+end
+
+# Thin plate spline approximation  
+# Based on:
+# Approximate Thin Plate Spline Mappings
+# by Gianluca Donato and Serge Belongie, 2002
```

Comment:
Is there a journal and DOI for this reference?

Reply:
Has been documented